packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -130,6 +130,16 @@ class RunCommand extends RunCommandBase {
         defaultsTo: true,
         help: 'If necessary, build the app before running.',
       )
+      ..addOption('dart-flags',
+        hide: !verboseHelp,
+        help: 'Pass a list of comma separated flags to the Dart instance at '
+              'application startup. Flags passed through this option must be '
+              'present on the whitelist defined within the Flutter engine. If '
+              'a non-whitelisted flag is encountered, the process will be '
+              'terminated immediately.\n\n'
+              'This flag is only applied in debug and profile modes and is not '
+              'recommended for use by users who are not working on the framework '
+              'or related tooling.')
```

Comment:
If this is only for local experiments, then let's disable this flag in stable versions by gating on !FlutterVersion.instance.isStable.